lib/dspace-item.ts

```diff
@@ -0,0 +1,60 @@
+import { Bitstream, Metadata } from './dspace-schemas';
+
+/**
+ * Returns an array of metadata items that match the given key.
+ * @param metadataItems - List of metadata to search.
+ * @param key - The key for the metadata items to find.
+ *
+ * @returns An array with metadata items for the given key.
+ * If no metadata items match the key an empty array is returned.
+ */
+export const findMetadataItems = (
+  metadataItems: Metadata[],
+  key: string
+): Metadata[] => metadataItems.filter((m) => m.key === key);
+
+/**
+   * Finds the first bistream item that matches the given bundle name
+   * and mimetype.
+   *
+   * @param bitstreams - List of bitstream items to search.
+   * @param bundleName - The DSpace bitstream bundle name.
+   * @param mimeType - The mimeType of the bitstream item.
+   * @returns The bitstream item that matches the given
+   * bundle name and mime type. Returns undefined if no matching bitstream
+   * is found.
+   */
+export const findBitstreamItem = (
+  bitstreamItems: Bitstream[],
+  bundleName: string,
```

Comment:
This could be DRY'd up a little bit with:

```suggestion
  bundleName: Bitstream['bundleName'],
```

That way, the `bundleName` parameter type would always be the same type as the `Bitstream`'s `bundleName` property. You wouldn't have to worry about a mismatch.

Reply:
This function was removed since it wasn't necessary after using the lodash syntax for the find functions.